### PR TITLE
Allow listing and searching of RA second factors

### PR DESCRIPTION
**Pull request by rjkip**

### CODE REVIEW

**Comment by rjkip** [comment]
:bell: @DRvanR

File: src/Surfnet/StepupMiddlewareClient/Identity/Dto/RaSecondFactorSearchQuery.php [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    private $pageNumber;
+
+    /**
+     * @param string $institution
+     * @param int $pageNumber
+     */
+    public function __construct($institution, $pageNumber)
+    {
+        $this->assertNonEmptyString($institution, 'institution');
+        Assert\that($pageNumber)
```
Comment (DRvanR) [inline src/Surfnet/StepupMiddlewareClient/Identity/Dto/RaSecondFactorSearchQuery.php]: We've not been using Assert for this up to now, for consistency sake I'd rather not start using them now. A chore to refactor all `if(!is_*){ throw exception }` to asserts would be very welcome. Till then I'd opt for consistency.

**Comment by DRvanR** [comment]
Apart from the asserts, looking good!

File: src/Surfnet/StepupMiddlewareClient/Identity/Dto/RaSecondFactorSearchQuery.php (hunk 1 above)
Comment (rjkip) [inline src/Surfnet/StepupMiddlewareClient/Identity/Dto/RaSecondFactorSearchQuery.php]: As discussed IRL, asserts are used in other search queries.

**Pull request merged**